src/App.jsx

```diff
@@ -14,33 +16,88 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABET = 'Sort alphabetically';
+const SORT_FIELD_LENGTH = 'Sort by length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const Good = ({ good }) => <li data-cy="Good">{good}</li>;
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+function getPreparedGoods(goods, { sortField, reversed }) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reversed,
+  });
```

Comment:
Getting `visibleGoods` this way is inefficient for every re-render would cause the array to be sorted and processed again. It would be better to apply the sort upon clicking and store the resulting array in a state!
That is: `onClick = {() => setVisibleGoods(getPreparedGoods(...))}`

This would also work if `visibleGoods` was wrapped in a `React.useMemo` (but that is a more advanced hook)